### PR TITLE
LGA-154: Use postcodes.io for postcode lookup

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Replaces our postcodeinfo.service.gov.uk integration and the fallback uk-postcodes.com/api.php integration with postcodes.io.

The changes aims to:

- [x] test integrations through mocks in unit tests and through real service in integration tests
- [x] log unsuccessful postcode requests, whether they were client, server or network errors
- [x] provide a visible JSON schema for the responses expected from the application

## Where should the reviewer start?

Commit-by-commit, please.

## Any other changes that would benefit highlighting?

The biggest change is the separation of seeding categories/types and adviser data.

Previously, `./manage.py migrate` would use `laalaa/apps/advisers/fixtures/initial_advisers.json` to populate categories, types and adviser data. As it imports 14000+ rows, it takes about 30 seconds.

These 30 seconds happen every time `./manage.py test` runs as it always starts from a clean test database and we don't use any of the data from it.

I split this file into `initial_categories.json` and `initial_advisers.json`. The former runs on migrations but the latter runs on a **new** `seed` command.

**TL;DR**: Tests are now 30 seconds faster and can create adviser data on demand.

The summary of changes:

| Data for model | Previous "initial advisers" | New "initial advisers" | initial_categories | Populated with |
| --- | --- | --- | --- | --- |
| advisers.category | ✅ |  | ✅ | migrate |
| advisers.organisationtype | ✅ |  | ✅ | migrate |
| advisers.outreachtype | ✅ |  | ✅ | migrate |
| advisers.organisation | ✅| ✅ | | seed |
| advisers.location |  ✅| ✅ | | seed |
| advisers.office | ✅| ✅ | | seed |
| advisers.outreachservice |  ✅| ✅ | | seed |

### CODE REVIEW

File: laalaa/apps/advisers/tests/test_geocoder.py [hunk 1]
```diff
@@ -15,7 +16,16 @@ class GeocoderTest(unittest.TestCase):
 
     def setUp(self):
         self.postcode = 'sw1a1aa'
-        self.good_result = {'status':200,'result':[{"postcode":"SW1A 1AA","longitude":-0.141588,"latitude":51.501009}]}
+        self.good_result = {'status':200,'result':[{'postcode':'SW1A 1AA','longitude':-0.141588,'latitude':51.501009}]}
+
+
+    @responses.activate
+    def test_geocode_strips_spaces_before_calling_external_api(self):
```
Comment (rowlando) [inline laalaa/apps/advisers/tests/test_geocoder.py]: :bowtie:

File: laalaa/apps/advisers/geocoder.py [hunk 2]
```diff
@@ -20,22 +20,23 @@ def __init__(self, postcode, *args, **kwargs):
...
-    postcode.postcode = format_postcode(result['postcode'])
+    postcode.postcode = normalise_postcode(result['postcode'])
     postcode.longitude = result['longitude']
     postcode.latitude = result['latitude']
     return postcode
 
 
 def lookup_postcode(postcode):
-    raw = requests.get('http://api.postcodes.io/postcodes/?q={postcode}&limit=1'.format(postcode=postcode))
+    normalised_postcode = normalise_postcode(postcode)
```
Comment (rowlando) [inline laalaa/apps/advisers/geocoder.py]: :bowtie:
